File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.cc
#include <iostream>
using namespace std;

int main(){
    int N;
    cin >> N;
    
    int ans = 0;
    for (int k = 1; k * (k - 1) / 2 < N; k++){
        int num = N - k * (k - 1) / 2;
        if(num % k == 0 && num / k > 0) {
            ans++;
        }
    }
    
    cout << ans << "\n";
    return 0;
}

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
N = int(input())

l = r = 1
sm = 0
ans = 0
while r <= N + 1:
    if sm < N:
        sm += r
        r += 1
    elif sm > N:
        sm -= l
        l += 1
    else:
        ans += 1
        sm += r
        r += 1
print(ans)
